### PR TITLE
Create diffdockcsv

### PR DESCRIPTION
**Pull request by flowscience**
run diffdock using a .csv of protein-ligand (PDB-SMILE) pairs

### CODE REVIEW

**Comment by NiklasTR** [comment]
Hi @flowscience thank you for adding a PR! 
There are multiple reasons why this tool is not working 

- [ ] $(inputs.protein.filepath) instead of inputs.csv in the first argument line
- [ ] same problem in line 3 of the arguments
- [ ] I do not think you need --ligand $(inputs.csv.filepath)  in line 3 of the arguments

I think making these changes should let it run

```
{
    "class": "CommandLineTool",
    "name": "diffdock",
    "description": "Docking of small molecules to a protein",
    "baseCommand": ["/bin/bash", "-c"],
    "arguments": [
      "python datasets/esm_embedding_preparation.py --protein_ligand_csv $(inputs.protein.filepath) --out_file /outputs/prepared_for_esm.fasta;",
      "HOME=esm/model_weights python esm/scripts/extract.py esm2_t33_650M_UR50D /outputs/prepared_for_esm.fasta /outputs/esm2_output --repr_layers $(inputs.repr_layers.default) --include per_tok && cp -r /outputs/esm2_output data/esm2_output;",
      "python -m inference --protein_ligand_csv $(inputs.protein.filepath) --ligand $(inputs.csv.filepath) --out_dir /outputs --inference_steps $(inputs.inference_steps.default) --samples_per_complex $(inputs.samples_per_complex.default) --batch_size $(inputs.batch_size.default) --actual_steps $(inputs.actual_steps.default) --no_final_step_noise;",
      "cp $(inputs.protein.filepath) /outputs"
    ],

    "dockerPull": "ghcr.io/labdao/diffdock:main@sha256:b00432de73478d3da578e4a16ee669178828109f3c7bf9c58d44bb7514f68629",
    "gpuBool": true,
    "networkBool": true,
    "memoryGB": 12,
    "inputs": {
      "protein": {
        "type": "File",
        "glob": ["*.pdb"]
      },
      "csv": {
        "type": "File",
        "glob": ["*.csv"]
      },
      "repr_layers": {
        "type": "int",
        "default": "33"
      },
      "inference_steps": {
        "type": "int",
        "default": "20"
      },
      "samples_per_complex": {
        "type": "int",
        "default": "40"
      },
      "batch_size": {
        "type": "int",
        "default": "10"
      },
      "actual_steps": {
        "type": "int",
        "default": "18"
      }
    },
    "outputs": {
      "best_docked_small_molecule": {
        "type": "File",
        "item": "",
        "glob": ["index*/rank1.sdf"]
      },
      "all_docked_small_molecules": {
        "type": "Array",
        "item": "File",
        "glob": ["index*/rank*.sdf"]
      },
      "protein": {
        "type": "File",
        "item": "",
        "glob": ["*.pdb"]
      }
    }
}
```

File: testdata/binding/input_protein_ligand.csv
Comment (NiklasTR) [inline testdata/binding/input_protein_ligand.csv:1]: you will run into issues here with you paths - the correct path is /inputs/6agt.pdb if you are mounting the file directly to the docker volume.

(you made the assumption that the docker image on plex is "seeing" your file in the google colab workspace, but that is not the case automatically. Instead it is mounted under /inputs)

File: tools/diffdock/diffdockcsv [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
+    "class": "CommandLineTool",
+    "name": "diffdock",
+    "description": "Docking of small molecules to a protein",
+    "baseCommand": ["/bin/bash", "-c"],
+    "arguments": [
+      "python datasets/esm_embedding_preparation.py --protein_ligand_csv $(inputs.protein.filepath) --out_file /outputs/prepared_for_esm.fasta;",
```
Comment (NiklasTR) [inline tools/diffdock/diffdockcsv:7]: issues in line 1 and 3 as described above

**Review by NiklasTR — CHANGES_REQUESTED** [review]

**Comment by flowscience** [comment]
> csv

Thank you for catching those and suggesting the appropriate fix.

To clarify, when you said "$(inputs.protein.filepath) instead of inputs.csv in the first argument line" - you mean I should have used "inputs.tsv.filepath" instead of "inputs.protein.filepath", correct? I believe this makes sense and I understand why, but I want to confirm before resubmitting.

**Pull request closed without merging**